Instastories.py

```diff
@@ -44,6 +46,32 @@ def get_media(url, path, media_type, username):
         base64_media = (base64.b64encode(image.read()).decode("utf-8"), media_type, username)
         return base64_media
 
+def save_cached_id(username, new_cached_id):
+    cached_ids = get_cached_ids()
+    cached_ids[username] = new_cached_id
+    with open(DEFAULT_CACHED_IDS_PATH, "w+") as file:
+        json.dump(cached_ids, file)
+
+def get_cached_ids():
+    if not os.path.exists(DEFAULT_CACHED_IDS_PATH):
+        return {}
+    else:
+        with open(DEFAULT_CACHED_IDS_PATH, "r") as file:
+            return json.load(file)
+        
+def check_cached_ids(ids):
+    new_ids = []
+    for element in ids:
+        if element.isdigit(): new_ids.append(element)  # If input is a valid id appends it
+        else:  # Else if input is nickname (not digit)
+            if element in get_cached_ids(): new_ids.append(get_cached_ids()[element]) # If username is in cache appends it
```

Comment:
Only on this line you are opening/closing the file 2 times. Save get_cached_ids at the beginning of the function.